esmvalcore/preprocessor/_time.py

```diff
@@ -248,6 +248,36 @@ def get_time_weights(cube):
     return time_weights
 
 
+def _aggregate_time_fx(result_cube, source_cube):
+    time_dim = set(source_cube.coord_dims(source_cube.coord('time')))
+    if source_cube.cell_measures():
+        for measure in source_cube.cell_measures():
+            measure_dims = set(source_cube.cell_measure_dims(measure))
+            if time_dim.intersection(measure_dims):
+                logger.debug('Averaging time dimension in measure %s.',
+                             measure.var_name)
+                result_measure = da.mean(measure.core_data(),
+                                         axis=tuple(time_dim))
+                measure = measure.copy(result_measure)
+                measure_dims = tuple(measure_dims - time_dim)
+                result_cube.add_cell_measure(measure, measure_dims)
+
+    if source_cube.ancillary_variables():
+        for ancillary_var in source_cube.ancillary_variables():
+            ancillary_dims = set(
+                source_cube.ancillary_variable_dims(ancillary_var))
+            if time_dim.intersection(ancillary_dims):
+                logger.debug(
+                    'Averaging time dimension in ancillary variable %s.',
+                    ancillary_var.var_name)
+                result_ancillary_var = da.mean(ancillary_var.core_data(),
+                                               axis=tuple(time_dim))
+                ancillary_var = ancillary_var.copy(result_ancillary_var)
+                ancillary_dims = tuple(ancillary_dims - time_dim)
+                result_cube.add_ancillary_variable(ancillary_var,
+                                                   ancillary_dims)
+
```

Comment:
the operations are identical for both `cell_measures` and `ancillary_variables` - I'd recommend building a function to avoid code duplication

Reply:
just build the var and dims once and pass them at add_* step - no worries, might be a bit overkill, the code reads fine as is now actually